include/envoy/http/header_map.h

```diff
@@ -349,15 +342,15 @@ class HeaderEntry {
  * have:
  *
  * ContentLength() -> returns the header entry if it exists or nullptr.
- * insertContentLength() -> inserts the header if it does not exist, and returns a reference to it.
+ * appendContentLength(foo, ",") -> append the string foo using a comma as a delimiter.
+ * setReferenceContentLength(foo) -> sets the header value to the reference string foo.
```

Comment:
can we use a different example than ContentLength in the doc for this particular method? IMO we should (though not necessarily in this PR) provide helper methods for ContentLength and Status that use integers.

I think it would be pretty unusual to call `setReferenceContentLength`; I'd expect the `setReferenceXXX` form to be used when the value is one of a well-known set that we have pre-initialized values for, such as Content-Type.